graphx/src/main/scala/org/apache/spark/graphx/GraphOps.scala

```diff
@@ -420,8 +420,9 @@ class GraphOps[VD: ClassTag, ED: ClassTag](graph: Graph[VD, ED]) extends Seriali
    *
    * @see [[org.apache.spark.graphx.lib.PageRank$#run]]
    */
-  def staticPageRank(numIter: Int, resetProb: Double = 0.15): Graph[Double, Double] = {
-    PageRank.run(graph, numIter, resetProb)
+  def staticPageRank(numIter: Int, resetProb: Double = 0.15,
```

Comment:
I think this will unnecessarily break binary compatibility though. Typically that's resolved by using overloads rather than new optional params. However, that would mean a new method for at least the 3 you modified here. Would it not be applicable to the others? which compounds the issue.

How about just adding it to the 'with options' version, by overloading it further?

Reply:
Should I do the same I did for every method? or not breakingf GraphOps interface might be enough?

Reply:
I'm wondering if you can add this new option in just one method, the most full-featured one, rather than every overload.

Reply:
Well, I could skip the change in run method and call directly runWithOptions from the overloaded version of staticPageRank, but I think runWithOptions would definitively need the new parameter, and for staticPageRank I can either overload (as it is now in the latest commit) or add a new parameter.

Reply:
The problem is that adding a new parameter, even with default, means the existing method signature no longer exists in the bytecode. while we can make breaking changes for Spark 3, I don't know if this is worth it. Hence, trying to figure out where an overload works, without adding a large number of them.

Reply:
Ok, I get it, I think we need staticPageRank and runWithOptions with the new interface. But the runWithOptions has the complete logic so in order not to duplicate code we would need to abstract the inner loop to a private function. Would it break?

Reply:
Refactoring, private functions are all fine.

Reply:
Did some changes

Reply:
Let me know if the changes now are fine, thank you very much